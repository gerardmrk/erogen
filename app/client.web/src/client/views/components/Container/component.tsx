import * as React from "react";
import styles from "./styles.scss";

type Props = {};

type State = {};

export class Container extends React.PureComponent<Props, State> {
  public render() {
    // prettier-ignore
    return (
      <div className={styles.main}>
        {this.props.children}
      </div>
    );
  }
}

export default Container;
