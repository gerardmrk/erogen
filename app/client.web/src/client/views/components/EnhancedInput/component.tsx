import * as React from "react";
import { LocalProps } from ".";
import Form from "@client/views/components/ui-collections/Form";
import Input from "@client/views/components/ui-elements/Input";
import styles from "./component.styles.scss";

export type Props = LocalProps;

export type State = {};

export class EnhancedInput extends React.PureComponent<Props, State> {
  public constructor(props: Props) {
    super(props);
  }

  public render() {
    return <Form.Field></Form.Field>;
  }
}
