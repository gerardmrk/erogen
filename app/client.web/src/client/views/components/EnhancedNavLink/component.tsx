import * as React from "react";
import { LocalProps, StoreProps, DispatchProps } from ".";
import { NavLink, NavLinkProps } from "react-router-dom";

export type Props = LocalProps & StoreProps & DispatchProps & NavLinkProps;

export type State = {};

export class EnhancedNavLink extends React.PureComponent<Props, State> {
  private onClick = (e: React.MouseEvent<HTMLAnchorElement>): void => {
    if (e.metaKey || e.ctrlKey || !this.props.appHasUpdates) return;
    e.preventDefault();
    window.location =
      typeof this.props.to === "object"
        ? { ...window.location, ...this.props.to }
        : { ...window.location, pathname: this.props.to };
  };

  public render() {
    return <NavLink {...this.props} onClick={this.onClick} />;
  }
}

export default EnhancedNavLink;
