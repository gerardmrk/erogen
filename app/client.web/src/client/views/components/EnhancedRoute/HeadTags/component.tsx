import * as React from "react";
import { Helmet } from "react-helmet";

export type Props = {
  title?: string;
  description?: string;
  metaType?: string;
};

export type State = {};

export class HeadTags extends React.PureComponent<Props, State> {
  public render() {
    return (
      <Helmet>
        <title>{""}</title>
      </Helmet>
    );
  }
}

export default HeadTags;
