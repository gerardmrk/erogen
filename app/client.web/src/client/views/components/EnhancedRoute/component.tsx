import * as React from "react";
import { LocationDescriptorObject } from "history";
import { Route as BaseRoute, RouteComponentProps, Redirect } from "react-router-dom"; // prettier-ignore
import { LocalProps, StoreProps, DispatchProps } from ".";
import { LOGIN_PATH, DEFAULT_PRIVATE_PATH } from "@client/views/conf.routes"; // prettier-ignore

export type Props = LocalProps & StoreProps & DispatchProps;

export type State = {};

export class EnhancedRoute extends React.Component<Props, State> {
  private redirectTo: LocationDescriptorObject = {
    pathname: LOGIN_PATH,
    search: "",
    state: { from: "" },
  };

  public constructor(props) {
    super(props);

    const fromRoute =
      props.location && props.location.pathname
        ? `${props.location.pathname}${props.location.search || ""}`
        : DEFAULT_PRIVATE_PATH;

    this.redirectTo.state.from = fromRoute;
    this.redirectTo.search = `?from=${encodeURIComponent(fromRoute)}`;
  }

  private renderRoute = (rcprops: RouteComponentProps) => {
    const Component = this.props.component;
    return <Component {...rcprops} routes={this.props.routes} />;
  };

  public render() {
    const { path, exact, strict, guarded, isAuthenticated } = this.props;

    if (guarded && !isAuthenticated) {
      return <Redirect to={this.redirectTo} />;
    }

    return (
      <BaseRoute
        path={path}
        exact={exact}
        strict={strict}
        render={this.renderRoute}
      />
    );
  }
}

export default EnhancedRoute;
