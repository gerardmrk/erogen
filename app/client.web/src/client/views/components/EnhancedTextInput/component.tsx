import * as React from "react";
import Form from "@client/views/components/ui-collections/Form";
import Input from "@client/views/components/ui-elements/Input";
import styles from "./component.styles.scss";
import { debounce } from "@client/utils/debounce";

export type Props = {
  id: string;
  type?: string;
  label: string;
  loading?: boolean;
  inputType?: string;
  autoComplete?: string;
  initialValue?: string;
  error?: string | undefined;
  disabled?: boolean;
  forceDisplayError?: boolean;
  onChange: (value: string) => void;
};

export type State = {
  value: string;
  dirty: boolean;
  touched: boolean;
};

// prettier-ignore
export class EnhancedTextInput extends React.Component<Props, State> {
  public static defaultProps = {
    type: "text",
    initialValue: "",
  }

  private delayedOnChange: () => void;

  public constructor(props: Props) {
    super(props);

    this.state = {
      value: props.initialValue as string,
      dirty: false,
      touched: false,
    };

    this.delayedOnChange = debounce(() => {
      this.props.onChange(this.state.value)
    }, 550);
  }

  public componentDidUpdate(prevProps: Props, prevState: State) {
    if (prevProps.disabled !== this.props.disabled) {
      this.props.onChange(this.state.value);
    }
  }

  private onChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    this.setState({
      touched: true,
      dirty: true,
      value: e.currentTarget.value,
    }, this.delayedOnChange);
  };

  public render() {
    return (
      <Form.Field widths={'equal'}>
        <label htmlFor={this.props.id}>
          {this.props.label}
        </label>

        <Input
          id={this.props.id}
          fluid={true}
          type={this.props.type}
          autoComplete={this.props.autoComplete}
          value={this.state.value}
          onChange={this.onChange}
          error={!!this.props.error && (this.state.dirty || this.props.forceDisplayError)}
          disabled={this.props.disabled}
          loading={this.props.loading}
        />

        {!!this.props.error && (this.state.dirty || this.props.forceDisplayError) && (
          <div className={styles.invalidation}>
            <span>
              {this.props.error}
            </span>
          </div>
        )}
      </Form.Field>
    );
  }
}
