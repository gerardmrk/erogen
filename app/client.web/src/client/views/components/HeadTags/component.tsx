import * as React from "react";
import { Helmet } from "react-helmet-async";
import { LocalProps } from ".";

export type Props = LocalProps;

export type State = {};

export class HeadTags extends React.PureComponent<Props, State> {
  public constructor(props: Props) {
    super(props);
  }

  // prettier-ignore
  public render() {
    const { config } = this.props;
    const titleTemplate = this.props.title ? `${config.app.appName} | %s` : "%s";
    const appName = config.app.appName;
    const url = !this.props.path || this.props.path === "/" ? config.app.appUrl : `${config.app.appUrl}${this.props.path}`;
    const title = this.props.title || config.app.appName;
    const description = this.props.description || config.app.appDescription;
    const keywords = this.props.keywords || config.app.appKeywords || config.app.appName;
    const metaType = this.props.metaType || "website";
    const metaImageUrl = this.props.metaImgPath ? `${url}${this.props.metaImgPath}` : `${url}${config.app.appImagePath}`;
    const metaImageAlt = this.props.metaImgAlt || config.app.appName;
    const metaTwitterCardType = this.props.metaTwitterCardType || config.app.appTwitterCardType || "summary_large_image";

    return (
      <Helmet titleTemplate={titleTemplate}>
        {/* HTML5 */}
        <link href={url} rel="canonical" />
        <title>{title}</title>
        <meta name="description" content={description} />
        <meta name="keywords" content={keywords} />

        {/* **NON-SEO** */}
        <meta
          name="viewport"
          content="width=device-width, initial-scale=1, shrink-to-fit=no"
        />

        {/* Schema.org */}
        <meta itemProp="url" content={url} />
        <meta itemProp="name" content={title} />
        <meta itemProp="description" content={description} />
        <meta itemProp="image" content={metaImageUrl} />

        {/* Open Graph */}
        <meta name="og:url" content={url} />
        <meta name="og:site_name" content={appName} />
        <meta name="og:title" content={title} />
        <meta name="og:description" content={description} />
        <meta name="og:type" content={metaType} />
        <meta name="og:image" content={metaImageUrl} />
        <meta name="og:image:alt" content={metaImageAlt} />

        {/* Twitter */}
        <meta name="twitter:domain" content={config.app.appUrl} />
        <meta name="twitter:site" content={config.app.appTwitterHandle} />
        <meta name="twitter:creator" content={config.app.appTwitterHandle} />
        <meta name="twitter:url" content={url} />
        <meta name="twitter:title" content={title} />
        <meta name="twitter:description" content={description} />
        <meta name="twitter:image" content={metaImageUrl} />
        <meta name="twitter:image:alt" content={metaImageAlt} />
        <meta name="twitter:card" content={metaTwitterCardType} />
      </Helmet>
    );
  }
}
