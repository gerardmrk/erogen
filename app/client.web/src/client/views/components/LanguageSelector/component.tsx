import * as React from "react";
import styles from "./component.styles.scss";
import { LocalProps, StoreProps, DispatchProps } from ".";
import Dropdown, {
  DropdownProps,
  DropdownItemProps,
} from "@client/views/components/ui.modules/Dropdown";

type Props = LocalProps & StoreProps & DispatchProps;

type State = {};

export class LanguageSelector extends React.PureComponent<Props, State> {
  private options: DropdownItemProps[] = [];

  public constructor(props: Props) {
    super(props);

    this.options = props.config.app.supportedLanguages.map(([code, lang]) => ({
      key: code,
      text: lang,
      value: code,
    }));
  }

  private onLanguageChange = async (
    e: React.SyntheticEvent<HTMLElement, Event>,
    { value }: DropdownProps,
  ) => {
    await this.props.i18n.changeLanguage(value as string);
  };

  public render() {
    return (
      <div className={styles.main}>
        <Dropdown
          fluid={true}
          selection={true}
          options={this.options}
          onChange={this.onLanguageChange}
          value={this.props.i18n.language}
        />
      </div>
    );
  }
}
