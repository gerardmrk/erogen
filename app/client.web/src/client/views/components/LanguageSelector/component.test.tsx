// Only enable snapshotting when component has stabilised and is not
// expected to undergo much development changes in the future.
const SNAPSHOT_ENABLED = true;

import * as React from "react";
import { ReactTestRenderer, ReactTestInstance } from "react-test-renderer"; // prettier-ignore
import { createRenderer } from "@client/views/__fixtures__/create-renderer";

import { LanguageSelector } from "./component";
import { withConfig } from "@client/views/core/ConfigProvider";
import { withTranslation } from "@client/views/core/I18nProvider";

const Component = withTranslation()(withConfig(LanguageSelector));

describe("<LanguageSelector/>", () => {
  let renderer: ReactTestRenderer;
  let component: ReactTestInstance;

  beforeEach(async () => {
    renderer = await createRenderer()(<Component />);
    component = renderer.getInstance() as ReactTestInstance;
    await (component as any).componentDidMount();
  });

  if (SNAPSHOT_ENABLED) {
    test("snapshot", () => {
      expect(renderer.toJSON()).toMatchSnapshot();
    });
  }

  it("renders ok", () => {
    expect(component).toBeDefined();
  });
});
