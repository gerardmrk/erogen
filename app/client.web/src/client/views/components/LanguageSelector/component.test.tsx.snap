// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LanguageSelector/> snapshot 1`] = `
<div
  className="main"
>
  <div
    aria-expanded={false}
    className="ui fluid selection dropdown"
    onBlur={[Function]}
    onChange={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseDown={[Function]}
    role="listbox"
    tabIndex={0}
  >
    <div
      aria-live="polite"
      className="text"
      role="alert"
    />
    <i
      aria-hidden="true"
      className="dropdown icon"
      onClick={[Function]}
    />
    <div
      className="menu transition"
    >
      <div
        aria-checked={false}
        aria-selected={true}
        className="selected item"
        onClick={[Function]}
        role="option"
        style={
          Object {
            "pointerEvents": "all",
          }
        }
      >
        <span
          className="text"
        >
          English
        </span>
      </div>
    </div>
  </div>
</div>
`;
